File: src/misc/R-functions/jaccard_similarity.R
#' @title jaccard_similarity
#' 
#' @description Calculates Jaccard similarity between two vectors
#' 
#' @param x The vector to analyze
#' @param y The vector to analyze
#' @export
#' 
#' @return

jaccard_similarity <- function(x, y)
{
    intersection = length(intersect(x, y))
    union = length(x) + length(y) - intersection
    intersection / union
}

File: src/statistics/jaccard_similarity.jl
export jaccard_similarity

"""
    jaccard_similarity(x, y)

Calculate Jaccard similarity between two vectors.

# Arguments

- `x::AbstractVector`
- `y::AbstractVector`

# Returns

- `j::Float64`
"""
function jaccard_similarity(x::AbstractVector, y::AbstractVector)

    i = float(length(intersect(x, y)))
    u = length(x) + length(y) - i

    return i / u

end

